test/scenarios-periodics/el92-src@cncf-conformance.sh

```diff
@@ -14,83 +14,139 @@ prepare_hosts() {
+    ssh -p "${secondary_host_ssh_port}" "redhat@${secondary_host_ip}" ./configure-sec.sh \
+        "${primary_host_name}" "${primary_host_ip}" \
+        "${secondary_host_name}" "${secondary_host_ip}" || rc=$?
+    if [ ${rc} -ne 0 ] ; then
+        record_junit "prepare_hosts" "configure_secondary" "FAILED"
+        return ${rc}
+    fi
+    record_junit "prepare_hosts" "configure_secondary" "OK"
 
+    # Configure kubeconfig and host name resolution
     scp -P "${primary_host_ssh_port}" "redhat@${primary_host_ip}:/home/redhat/kubeconfig-${primary_host_name}" "${KUBECONFIG}"
     export KUBECONFIG="${KUBECONFIG}"
     echo "${primary_host_ip} ${primary_host_name}" | sudo tee -a /etc/hosts &>/dev/null
+
+    return ${rc}
 }
 
 run_sonobuoy() {
+    local rc=0
+    local start
+    local now
+
     # Configure cluster prerequisites
-    oc adm policy add-scc-to-group privileged system:authenticated system:serviceaccounts
-    oc adm policy add-scc-to-group anyuid     system:authenticated system:serviceaccounts
+    oc adm policy add-scc-to-group privileged system:authenticated system:serviceaccounts || rc=$?
+    oc adm policy add-scc-to-group anyuid     system:authenticated system:serviceaccounts || rc=$?
+    if [ ${rc} -ne 0 ] ; then
+        record_junit "run_sonobuoy" "add_scc_to_group" "FAILED"
+        return ${rc}
+    fi
 
+    # Initiate test startup
     go install github.com/vmware-tanzu/sonobuoy@v0.57.1
     ~/go/bin/sonobuoy run \
         --mode=certified-conformance \
         --dns-namespace=openshift-dns \
         --dns-pod-labels=dns.operator.openshift.io/daemonset-dns=default \
-        --e2e-parallel=y 
+        --e2e-parallel=y || rc=$?
+    if [ ${rc} -ne 0 ] ; then
+        record_junit "run_sonobuoy" "start_e2e" "FAILED"
+        return ${rc}
+    fi
+    record_junit "run_sonobuoy" "start_e2e" "OK"
 
     # Wait for up to 5m until tests start
-    WAIT_FAILURE=true
+    rc=1
     for _ in $(seq 1 150) ; do
         if [ "$(~/go/bin/sonobuoy status --json | jq -r '.status')" = "running" ]; then
-            WAIT_FAILURE=false
+            rc=0
             break
         fi
         sleep 2
     done
-    if ${WAIT_FAILURE}; then
+    if [ ${rc} -ne 0 ] ; then
         echo "Failed to start tests after 5m"
-        exit 1
+        record_junit "run_sonobuoy" "wait_e2e_running" "FAILED"
+        return ${rc}
     fi
+    record_junit "run_sonobuoy" "wait_e2e_running" "OK"
 
     # Use 1h timeout. A normal run on 2 CPUs takes 40-45min.
     start=$(date +%s)
     while [ "$(~/go/bin/sonobuoy status --json | jq -r '.status')" = "running" ] ; do
         now=$(date +%s)
         if [ $(( now - start )) -ge ${TIMEOUT_TEST} ]; then
+            rc=1
             echo "Tests running for 1h. Timing out"
+            record_junit "run_sonobuoy" "wait_e2e_finished" "FAILED"
```

Comment:
Does the full status output include any information useful to debug the tests? Should we copy it to the artifacts directory?

Reply:
Good idea. Will do.